src/main/java/net/minecraftforge/client/model/pipeline/VertexLighterFlat.java

```diff
@@ -233,18 +233,19 @@ protected void updateLightmap(float[] normal, float[] lightmap, float x, float y
     {
         float e1 = 1 - 1e-2f;
         float e2 = 0.95f;
-        BlockPos pos = blockInfo.getBlockPos();
 
-        boolean full = blockInfo.getState().isFullCube();
+        boolean full = blockInfo.isFullCube();
+        EnumFacing side = null;
 
-        if((full || y < -e1) && normal[1] < -e2) pos = pos.down();
-        if((full || y >  e1) && normal[1] >  e2) pos = pos.up();
-        if((full || z < -e1) && normal[2] < -e2) pos = pos.north();
-        if((full || z >  e1) && normal[2] >  e2) pos = pos.south();
-        if((full || x < -e1) && normal[0] < -e2) pos = pos.west();
-        if((full || x >  e1) && normal[0] >  e2) pos = pos.east();
+             if((full || y < -e1) && normal[1] < -e2) side = EnumFacing.DOWN;
+        else if((full || y >  e1) && normal[1] >  e2) side = EnumFacing.UP;
```

Comment:
Are you sure the if...else transformation is correct here? I think it will only hold true for the same axis, and you could have, say, moving the position in both the Y and Z axis.

Reply:
The normal vectors here are supposed to have unit length. a value > 0.95 in a particular direction precludes a similar value in other directions.

Reply:
Okay, thanks!